src/main/java/uk/co/fivium/dmda/server/SMTPConfig.java

```diff
@@ -236,13 +250,15 @@ else if (lDomainRegexNode != null && lDomainNode == null) {
             }
           }
 
-          lRecipientDatabaseMap.add(new DomainMatcher(lDomain, lIsRegexDomain, lDatabaseName));
+          lRecipientDatabaseMap.add(new DomainMatcher(lDomain, lIsRegexDomain, lDatabaseName, lPriority));
         }
         else {
           throw new ConfigurationException("Invalid recipient XML");
         }
       }
 
+      lRecipientDatabaseMap.sort(Comparator.nullsLast(Comparator.comparing(DomainMatcher::getPriority)));
```

Comment:
Does `Comparator.nullsLast` actually do anything here given that `getPriority` gets coalesced to `Integer.MAX_VALUE` above anyway?